tests/framework/validators/DateValidatorTest.php

```diff
@@ -602,4 +602,19 @@ public function testTimestampAttributeSkipValidation()
         $val->validateAttribute($model, 'attr_date');
         $this->assertTrue($model->hasErrors('attr_date'));
     }
+
+    /**
+     * @depends testValidateAttributePHPFormat
+     */
+    public function testTimestampAttributeEmptyValueValidation()
+    {
```

Comment:
this method should cover more scenario's to cover the `if` statement you've added